File: Bikes.Domain/Bike.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Bikes.Domain
{
    public enum BikeSize { Small, Medium, Large }
    public enum BikeType { Mountain, Road, Hybrid, Electric }

    public class Bike
    {
        // Klucz główny
        public int BikeId { get; set; } // Auto-incremented primary key

        // Podstawowe właściwości
        public string Name { get; set; }
        public BikeSize Size { get; set; }
        public BikeType BikeType { get; set; }
        public bool IsElectric { get; set; }
        public decimal HourlyRate { get; set; }
        public decimal DailyRate { get; set; }
        public bool AvailabilityStatus { get; set; }

        // Nawigacja
        [JsonIgnore] // Ignorowanie, aby uniknąć cyklicznych referencji
        public ICollection<Reservation> Reservations { get; set; }

        [JsonIgnore]
        public ICollection<Feedback> Feedbacks { get; set; }

        [JsonIgnore]
        public ICollection<Damage> Damages { get; set; }

        [JsonIgnore]
        public ICollection<Availability> Availabilities { get; set; }
    }
}

File: Bikes.API/clientapp/src/Models/Bike.ts
export enum BikeSize {
    Small = 'Small',
    Medium = 'Medium',
    Large = 'Large',
}

export enum BikeType {
    Mountain = 'Mountain',
    Road = 'Road',
    Hybrid = 'Hybrid',
    Electric = 'Electric',
}

export interface Reservation {
    id: string;
    startDate: string; 
    endDate: string; 
    bikeId: string;
}

export interface Feedback {

    id: string;
    bikeId: string;
    rating: number; 
    comment: string;
}

export interface Damage {

    id: string;
    description: string;
    dateReported: string; 
    bikeId: string;
}

export interface Availability {
    id: string;
    bikeId: string;
    availableFrom: string; 
    availableTo: string;
}

export interface Bike {
    id: string; 
    name: string;
    size: BikeSize;
    type: BikeType; 
    isElectric: boolean;
    hourlyRate: number;
    dailyRate: number;
    availabilityStatus: boolean;

}

export default Bike;